### PR TITLE
[#IOCOM-921] new function GetRCConfiguration

### PR DESCRIPTION
**Pull request by acuiuli**
#### List of Changes

- [new function GetRCConfiguration](https://github.com/pagopa/io-functions-app-messages/commit/e503a4e7a3174b6850b01d024466b3c83f889eaf)

#### Motivation and Context

Created a new function to retrieve the remote content configuration by ID

#### How Has This Been Tested?

Unit test

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: GetRCConfiguration/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+secureExpressApp(app);
+
+const config = getConfigOrThrow();
+
+const rCConfigurationModel = new RCConfigurationModel(
+  cosmosdbInstance.container(RC_CONFIGURATION_COLLECTION_NAME)
+);
+
+app.get(
+  "v1/remote-contents/configurations/{id}",
```
Comment (Garma00) [inline GetRCConfiguration/index.ts]: The path should start with `/api` [for example](https://github.com/pagopa/io-functions-app-messages/blob/master/GetMessage/index.ts#L56)

File: GetRCConfiguration/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { secureExpressApp } from "@pagopa/io-functions-commons/dist/src/utils/express";
+import { setAppContext } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+
+import createAzureFunctionHandler from "@pagopa/express-azure-functions/dist/src/createAzureFunctionsHandler";
+
+import {
+  RCConfigurationModel,
+  RC_CONFIGURATION_COLLECTION_NAME
+} from "@pagopa/io-functions-commons/dist/src/models/rc_configuration";
+import { cosmosdbInstance } from "../utils/cosmosdb";
```
Comment (Garma00) [inline GetRCConfiguration/index.ts]: Take care, here we want to use the cosmos instance for remote content see [here](https://github.com/pagopa/io-functions-app-messages/blob/master/utils/cosmosdb.ts#L32)

File: GetRCConfiguration/handler.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    ),
+    TE.map(ResponseSuccessJson),
+    TE.toUnion
+  )();
+
+/**
+ * Wraps a GetRCConfiguration handler inside an Express request handler.
+ */
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions, max-params
+export function GetRCConfiguration(
```
Comment (Garma00) [inline GetRCConfiguration/handler.ts]: What about using an arrow function here and remove the eslint-disable?
Also i would suggest to stat the name of the function without the capital "G" for convention

File: GetRCConfiguration/handler.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ */
+type IGetRCConfigurationHandler = (
+  context: Context,
+  configurationId: Ulid
+) => Promise<IGetRCConfigurationHandlerResponse>;
+
+/**
+ * Handles requests for getting a single remote content configuration for the requested id.
+ */
+export const GetRCConfigurationHandler = (
```
Comment (Garma00) [inline GetRCConfiguration/handler.ts:52]: I would start the name without the capital "G" for convention

File: utils/remoteContentConfig.ts [hunk 5]
```diff
@@ -79,3 +87,65 @@ export const getOrCacheRemoteServiceConfig = (
...
+                `REMOTE-CONTENT-CONFIGURATION-${configurationId}`,
+                JSON.stringify(rCConfiguration),
+                rCConfigurationCacheTtl
+              ),
+              TE.map(() => rCConfiguration),
+              TE.orElse(() => TE.of(rCConfiguration))
+            )
+          )
+        ),
+      x => TE.right(O.some(x))
```
Comment (Garma00) [inline utils/remoteContentConfig.ts]: I would use something like `remoteConfiguration` to rename `x`

**Review by Garma00 — CHANGES_REQUESTED** [review]
Well done, just some comments 😃

File: GetRCConfiguration/handler.ts (hunk 4 above)
Comment (acuiuli) [reply GetRCConfiguration/handler.ts:52]: All the other handlers of functions in this repo also start with a capital letter, so I followed it. Check it [here](https://github.com/pagopa/io-functions-app-messages/blob/33ab4864da15fbb500542d957bd07fbc270fe112/GetMessages/handler.ts#L63C14-L63C32) and [here](https://github.com/pagopa/io-functions-app-messages/blob/33ab4864da15fbb500542d957bd07fbc270fe112/GetMessage/handler.ts#L142)

**Review by michaeldisaro — APPROVED** [review]
LGTM

**Review by Garma00 — APPROVED** [review]
lgtm

**Pull request merged**